src/main/scala/io/sdkman/changelogs/package.scala

```diff
@@ -200,7 +224,9 @@ package object changelogs {
       version: String,
       url: String,
       platform: Platform = Universal,
-      vendor: Option[Vendor] = None
+      vendor: Option[Vendor] = None,
+      hashAlgorithm: Option[HashAlgorithm] = None,
+      checksum: Option[String] = None
```

Comment:
I'm not sure if that entirely makes sense. Can a checksum exist without an algorithm? These two together should probably have a case class of their own, and that should be an option.
Something like:
```
case class Checksum(algorithm: HashAlgorithm, value: String)
...
checksum: Option[Checksum]
```

Reply:
Agreed. But if this is slowing down the pull request, I'd be to contribute an edit.

Reply:
@stewSquared perhaps you aren't getting the context here, but you are looking at the database migration project. This will be the last thing to be implemented. It is also the least important of the lot.

We are driving this feature from the client side, and are currently focusing on the CLI. After that we will do the crucial work in the broker, followed by all the other auxillary services.

At the moment, everything works on the client, with the exception of one small bug that we've encountered with cached archives. The fix will be released soon. Checksums will be released into the wild when we think it's ready, not a moment sooner.